### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by desilinguist**
1. Add sections and table of contents.
2. Add much more detail about each command and what it is doing.
3. Add Travis CI badge.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,87 +1,135 @@
...
 
-```
-extract_segmentation_features rst_discourse_tb_edus_TRAINING_TRAIN.json rst_discourse_tb_edus_features_TRAINING_TRAIN.tsv
+    ```bash
+    conda activate rstenv
+    ```
 
-extract_segmentation_features rst_discourse_tb_edus_TRAINING_DEV.json rst_discourse_tb_edus_features_TRAINING_DEV.tsv
-```
+2. **Download & setup tokenization & syntactic parsing models**. We need to download some data
```
Comment (ghost) [inline README.md]: this bullet seems incomplete.

File: README.md [hunk 2]
```diff
@@ -1,87 +1,135 @@
...
-```
-rst_parse -g segmentation_model.C1.0 -p rst_parsing_model.C1.0 my_document
-```
+    - `rst_discourse_tb_edus_TRAINING.json` : the original RST Discourse Treebank training set merged with the corresponding Penn Treebank trees in JSON format. 
+
+    - `rst_discourse_tb_edus_TEST.json` : the original RST Discourse Treebank test set merged with the corresponding Penn Treebank trees in JSON format. 
+
+    - `rst_discourse_tb_edus_TRAINING_DEV.json`: the development set split out from the original training set. This fill will be used to tune the segmenter and RST parser hyperparameters. 
+
+    - `rst_discourse_tb_edus_TRAINING_TRAIN.json` : a smaller training set split out from the original training set. This file will be used to train the segmenter and the parser.
```
Comment (ghost) [inline README.md]: Instead of saying split out from original training set, will it be clearer to say `rst_discourse_tb_edus_TRAINING_TRAIN.json` and `rst_discourse_tb_edus_TRAINING_DEV.json` are train/dev splits from `rst_discourse_tb_edus_TRAINING.json` ?

File: README.md [hunk 3]
```diff
@@ -1,87 +1,135 @@
...
-```
+    - `rst_discourse_tb_edus_TRAINING.json` : the original RST Discourse Treebank training set merged with the corresponding Penn Treebank trees in JSON format. 
+
+    - `rst_discourse_tb_edus_TEST.json` : the original RST Discourse Treebank test set merged with the corresponding Penn Treebank trees in JSON format. 
+
+    - `rst_discourse_tb_edus_TRAINING_DEV.json`: the development set split out from the original training set. This fill will be used to tune the segmenter and RST parser hyperparameters. 
+
+    - `rst_discourse_tb_edus_TRAINING_TRAIN.json` : a smaller training set split out from the original training set. This file will be used to train the segmenter and the parser. 
+
+4. **Extract the segmenter features**. Create inputs (features and labels) for training a discourse segmentation model for the newly created training set:
```
Comment (ghost) [inline README.md]:
```suggestion
4. **Extract the segmenter features**. Create inputs (features and labels) to train a discourse segmentation model from the newly created training set:
```

File: README.md [hunk 4]
```diff
@@ -1,87 +1,135 @@
...
+    and the development set:
 
-To evaluate an existing model, run:
+    ```bash
+    extract_segmentation_features rst_discourse_tb_edus_TRAINING_DEV.json rst_discourse_tb_edus_features_TRAINING_DEV.tsv
+    ```
 
+    The extracted features for the training and development set are now in the `rst_discourse_tb_edus_features_TRAINING_TRAIN.tsv` and `rst_discourse_tb_edus_features_TRAINING_DEV.tsv` files respectively.
+
+5. **Train the segmenter CRF model and tune its hyper-parameters**. Train (with the training set) and tune (with the development set) a CRF-based discourse segmentation model:
```
Comment (ghost) [inline README.md]:
```suggestion
5. **Train the CRF segmenter  model and tune its hyper-parameters**. Train (with the training set) and tune (with the development set) a CRF-based discourse segmentation model:
```

File: README.md [hunk 5]
```diff
@@ -1,87 +1,135 @@
...
+    ```
+
+    This command iterates over a pre-defined list of values for the `C` regularization parameter for logistic regression, trains a model using the features extracted from the training set, and then evaluates that model on the development set. Its final output is the `C` value that yields the highest performance F1 score on the development set. After this command, you will have a number of directories with the prefix `rst_parsing_model` in the current directory, e.g., `rst_parsing_model.C0.25`, `segmentation_model.C1.0` et cetera. Each of these directories contains the logistic regression model files (named `rst_parsing_all_feats_LogisticRegression.model`) trained with those specific values of the `C` regularization parameter.  Underlyingly, this command uses the [SKLL](https://skll.readthedocs.io) machine learning library to train and evaluate the models.
+
+7. (Optional) **Evaluate trained model**. If you want to obtain detailed evaluation metrics for an RST parsing model, run:
+
+    ```bash
+    rst_eval rst_discourse_tb_edus_TRAINING_DEV.json -p rst_parsing_modelC1.0 --use_gold_syntax
```
Comment (ghost) [inline README.md]:
```suggestion
    rst_eval rst_discourse_tb_edus_TEST.json -p rst_parsing_modelC1.0 --use_gold_syntax
```

File: README.md [hunk 6]
```diff
@@ -1,87 +1,135 @@
...
+### Use trained models
+
+At this point, we are ready to use the segmentation and RST parsing models to process raw text documents. To process a raw text document `document.txt` with the end-to-end parser (assuming `C` = 1.0 was the best hyper-parameter value for both the segmentation and RST parsing models), run:
+
+```bash
+rst_parse -g segmentation_model.C1.0 -p rst_parsing_model.C1.0 document.txt > output.json
 ```
-rst_eval rst_discourse_tb_edus_TRAINING_DEV.json -p rst_parsing_modelC1.0 --use_gold_syntax
+
+`output.json` contains a dictionary with two keys: `edu_tokens` and `scored_rst_trees`. The value corresponding to `edu_tokens` is a list of lists: each list is a list of tokens containing the segmented Elementary Discourse Units (EDUs). The value corresponding to `rst_trees` is a list of dictionaries: each dictionary has two keys, `tree` and `score` containing the RST parse tree for the document and its score respectively. By default, only a single tree is produed but additonal trees can be produced by specifying the `-n` option for `rst_parse`.
```
Comment (ghost) [inline README.md]: each list is a list of tokens containing the segmented Elementary Discourse Units (EDUs). -> each list is a list of tokens in a segmented Elementary Discourse Units (EDUs).

**Review by ghost — COMMENTED** [review]
It looks great! I have a few minor comments.

File: README.md (hunk 1 above)
Comment (desilinguist) [reply README.md]: Oh, good catch! I totally forgot about this!

File: README.md (hunk 2 above)
Comment (desilinguist) [reply README.md]: Yep, that will be clearer. I'll do that.

File: README.md (hunk 6 above)
Comment (desilinguist) [reply README.md]: Hmm, I think even this is not clear enough. Let me rewrite this a bit.

File: README.md (hunk 5 above)
Comment (desilinguist) [reply README.md]: No, we actually don't use the test set at all but I guess we could.

File: README.md [hunk 7]
```diff
@@ -1,87 +1,151 @@
...
+    ```
+
+    This command iterates over a pre-defined list of values for the `C` regularization parameter for logistic regression, trains a model using the features extracted from the training set, and then evaluates that model on the development set. Its final output is the `C` value that yields the highest performance F1 score on the development set. After this command, you will have a number of directories with the prefix `rst_parsing_model` in the current directory, e.g., `rst_parsing_model.C0.25`, `segmentation_model.C1.0` et cetera. Each of these directories contains the logistic regression model files (named `rst_parsing_all_feats_LogisticRegression.model`) trained with those specific values of the `C` regularization parameter.  Underlyingly, this command uses the [SKLL](https://skll.readthedocs.io) machine learning library to train and evaluate the models.
+
+7. (Optional) **Evaluate trained model**. If you want to obtain detailed evaluation metrics for an RST parsing model on the development set, run:
+
+    ```bash
+    rst_eval rst_discourse_tb_edus_TRAINING_DEV.json -p rst_parsing_modelC1.0 --use_gold_syntax
```
Comment (ghost) [inline README.md]:
```suggestion
    rst_eval rst_discourse_tb_edus_TRAINING_DEV.json -p rst_parsing_model.C1.0 --use_gold_syntax
```

**Review by ghost — DISMISSED** [review]

File: README.md (hunk 7 above)
Comment (desilinguist) [reply README.md]: Good catch!

**Review by aoifecahill — APPROVED** [review]
👍

**Review by ghost — APPROVED** [review]
